File: src/core/browser/browser_factory.py
from src.common import Factory

from .browser_manager import BrowserManager
from .browser_models import BrowserConfig
from .imp.playwright_browser_manager import PlaywrightBrowserManager


class BrowserManagerFactory(Factory[BrowserManager]):
    def create(self, config: object) -> BrowserManager:
        if isinstance(config, BrowserConfig):
            return PlaywrightBrowserManager(config)
        return PlaywrightBrowserManager(BrowserConfig())

File: src/wheel/browser/browser_factory.cpp
#include "wheel/browser/browser_factory.h"
#include "wheel/browser/imp/cdp_browser_manager.h"

namespace lynne {
namespace wheel {

BrowserManager* BrowserFactory::create(const BrowserConfig& config) const {
    return new CdpBrowserManager(config);
}

} // namespace wheel
} // namespace lynne
